src/org/jetbrains/plugins/scala/caches/HydraArtifactsCache.scala

```diff
@@ -0,0 +1,51 @@
+package org.jetbrains.plugins.scala.caches
+
+import java.io.File
+import java.util.concurrent.ConcurrentHashMap
+
+import com.intellij.openapi.diagnostic.Logger
+import org.jetbrains.plugins.scala.project.template.Downloader
+
+/**
+  * @author Maris Alexandru
+  */
+object HydraArtifactsCache {
+  private val LOG = Logger.getInstance(this.getClass)
+  private val cache = new ConcurrentHashMap[(String, String), Seq[String]]()
+  private val SplitRegex = "\\* Attributed\\(|\\)".r
+  private val ArtifactsRegex = "\\* Attributed\\(.*\\)".r
+  private val GroupId = "com.triplequote"
+  private val PathSeparator = File.separator
+
+  def getOrDownload(scalaVersion: String, hydraVersion: String, listener: (String) => Unit): Seq[String] = {
+    val artifacts = cache.getOrDefault((scalaVersion, hydraVersion), Seq.empty)
+
+    if (artifacts.isEmpty) {
+      val stringBuilder = new StringBuilder
```

Comment:
Please update the name of this variable to make it clear how it's used. I'd rename it to `sbtBufferedOutput` or similar